pkg/tests/apis/provisioning/provisioning_test.go

```diff
@@ -92,7 +86,41 @@ func TestIntegrationProvisioning(t *testing.T) {
 		},
 	})
 
+	cleanSlate := func(t *testing.T) {
```

Comment:
This new function is not strictly necessary for all subtests, but it is very helpful where you don't want to be bothered by impurities.

I don't know of any way to make `testing.T` do clean up on subtests on its own beyond just calling `t.Cleanup(cleanSlate)`, at which point I figured we might as well just call it directly instead.